

<!DOCTYPE html>


<html lang="en" data-content_root="" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>Cell Painting Gallery folder structure &#8212; Cell Painting Gallery Documentation</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=5b4479735964841361fd" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=5b4479735964841361fd" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=5b4479735964841361fd" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd" />
  <script src="_static/vendor/fontawesome/6.1.2/js/all.min.js?digest=5b4479735964841361fd"></script>

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script async="async" kind="hypothesis" src="https://hypothes.is/embed.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'data_structure';</script>
    <link rel="canonical" href="broadinstitute.github.io/data_structure.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Feature Sets" href="feature_sets.html" />
    <link rel="prev" title="Cell Painting Gallery Overview" href="overview.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <div id="pst-scroll-pixel-helper"></div>

  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>
    Back to top
  </button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  

<a class="navbar-brand logo" href="overview.html">
  
  
  
  
  
  
    <p class="title logo__title">Cell Painting Gallery Documentation</p>
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="overview.html">
                    Cell Painting Gallery Overview
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Data</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Cell Painting Gallery folder structure</a></li>
<li class="toctree-l1"><a class="reference internal" href="feature_sets.html">Feature Sets</a></li>
<li class="toctree-l1"><a class="reference internal" href="browsing_data.html">Browsing Data on the Cell Painting Gallery</a></li>
<li class="toctree-l1"><a class="reference internal" href="download_instructions.html">Downloading from Cell Painting Gallery</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Applications</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="workflows.html">Workflows using the Cell Painting Gallery</a></li>
<li class="toctree-l1"><a class="reference internal" href="machine_learning.html">Machine Learning (ML)/Deep Learning (DL)</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Technical Guides</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="citing.html">Citing the Cell Painting Gallery</a></li>
<li class="toctree-l1"><a class="reference internal" href="contributing_to_cpg.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="uploading_to_cpg.html">Tutorial: Uploading to the Cell Painting Gallery</a></li>
<li class="toctree-l1"><a class="reference internal" href="prefixes.html">Prefixes</a></li>
<li class="toctree-l1"><a class="reference internal" href="publications.html">Publications using datasets in Cell Painting Gallery</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/broadinstitute/cellpainting-gallery" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/broadinstitute/cellpainting-gallery/edit/main/documentation/data_structure.md" target="_blank"
   class="btn btn-sm btn-source-edit-button dropdown-item"
   title="Suggest edit"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-pencil-alt"></i>
  </span>
<span class="btn__text-container">Suggest edit</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/broadinstitute/cellpainting-gallery/issues/new?title=Issue%20on%20page%20%2Fdata_structure.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/data_structure.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Cell Painting Gallery folder structure</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#images-folder-structure"><code class="docutils literal notranslate"><span class="pre">images</span></code> folder structure</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#arrayed-cell-painting-experiments">Arrayed Cell Painting experiments</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#pooled-cell-painting-experiments">Pooled Cell Painting experiments</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#workspace-folder-structure"><code class="docutils literal notranslate"><span class="pre">workspace</span></code> folder structure</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#analysis-folder-structure"><code class="docutils literal notranslate"><span class="pre">analysis</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#backend-folder-structure"><code class="docutils literal notranslate"><span class="pre">backend</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-data-csv-folder-structure"><code class="docutils literal notranslate"><span class="pre">load_data_csv</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#metadata-folder-structure"><code class="docutils literal notranslate"><span class="pre">metadata</span></code> folder structure</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#arrayed-metadata">arrayed metadata</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#pooled-metadata">pooled metadata</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#profiles-folder-structure"><code class="docutils literal notranslate"><span class="pre">profiles</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#quality-control-folder-structure"><code class="docutils literal notranslate"><span class="pre">quality_control</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#segmentation-folder-structure"><code class="docutils literal notranslate"><span class="pre">segmentation</span></code> folder structure</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#workspace-dl-folder-structure"><code class="docutils literal notranslate"><span class="pre">workspace_dl</span></code> folder structure</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#collated-folder-structure"><code class="docutils literal notranslate"><span class="pre">collated</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#consensus-folder-structure"><code class="docutils literal notranslate"><span class="pre">consensus</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#embeddings-folder-structure"><code class="docutils literal notranslate"><span class="pre">embeddings</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1"><code class="docutils literal notranslate"><span class="pre">profiles</span></code> folder structure</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#complete-folder-structure">Complete folder structure</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section id="cell-painting-gallery-folder-structure">
<h1>Cell Painting Gallery folder structure<a class="headerlink" href="#cell-painting-gallery-folder-structure" title="Permalink to this heading">#</a></h1>
<p>All projects in the Cell Painting Gallery form a stereotyped structure.
The parent structure is as follows.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cellpainting-gallery
└── &lt;project&gt;
    └── &lt;project-specific-nesting&gt;
        ├── images
        ├── workspace
        └── workspace_dl
</pre></div>
</div>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">&lt;project&gt;</span></code>: top level folder for the project.
Keep the name short and simple with <code class="docutils literal notranslate"><span class="pre">[a-z0-9_]</span></code> only.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">&lt;project-specific-nesting&gt;</span></code>: additional nesting level that is typically an institution identifier.
It should be present even if the data is from a single source (e.g. <code class="docutils literal notranslate"><span class="pre">s3://cellpainting-gallery/cpg0003-rosetta/</span></code> only contains <code class="docutils literal notranslate"><span class="pre">broad/</span></code>).
It can be anonymized (e.g. <code class="docutils literal notranslate"><span class="pre">s3://cellpainting-gallery/cpg0016-jump/</span></code> contains <code class="docutils literal notranslate"><span class="pre">source_1/</span></code>, <code class="docutils literal notranslate"><span class="pre">source_2/</span></code>, etc.).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">images</span></code>: all images and illumination correction functions.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">workspace</span></code>: everything else that results from CellProfiler-based features goes here.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">workspace_dl</span></code>: everything else that results from deep learning-based features goes here.</p></li>
</ul>
<p>Not all projects will have all parent structures.</p>
<p>The “completeness” of a project can be checked using our <a class="reference external" href="https://github.com/broadinstitute/cpg">data validator</a>.
Please note that it is in alpha and furhter functionality and documentation are under development.</p>
<section id="images-folder-structure">
<h2><code class="docutils literal notranslate"><span class="pre">images</span></code> folder structure<a class="headerlink" href="#images-folder-structure" title="Permalink to this heading">#</a></h2>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cellpainting-gallery/
└── &lt;project&gt;
    └── &lt;project-specific-nesting&gt;
        └── images
        │   ├── YYYY_MM_DD_&lt;batch-name&gt;
        │   │   ├── illum
        │   │   │   ├── &lt;plate-name&gt;
        │   │   │   │   ├── &lt;plate-name&gt;_Illum&lt;Channel&gt;.npy
        │   │   │   │   └── &lt;plate-name&gt;_Illum&lt;Channel&gt;.npy
        │   │   │   └── &lt;plate-name&gt;
        │   │   └── images
        │   │       ├── &lt;full-plate-name&gt;
        │   │       └── &lt;full-plate-name&gt;
        │   └── YYYY_MM_DD_&lt;batch-name&gt;
        └── workspace
</pre></div>
</div>
<p>Within the outer <code class="docutils literal notranslate"><span class="pre">images</span></code> folder, there are <code class="docutils literal notranslate"><span class="pre">YYYY_MM_DD_&lt;batch-name&gt;</span></code> subfolders for each batch.
Each batch folder typically starts with <code class="docutils literal notranslate"><span class="pre">YYYY_MM_DD</span></code> of the date that image acquisition started.
The rest of the batch folder name can be a simple ordinal (e.g. <code class="docutils literal notranslate"><span class="pre">YYYY_MM_DD_Batch1</span></code>) or more descriptive of its contents (e.g. <code class="docutils literal notranslate"><span class="pre">2020_01_02_TestPhalloidinConcentration</span></code>).
A single batch typically contains all of the plates that were imaged (or started acquisition) on that day.
However, for simplifying project tracking and analysis, sometimes plates imaged on the same day are divided into multiple batches where each batch is a different experimental condition (e.g. <code class="docutils literal notranslate"><span class="pre">2020_01_02_LowPhalloidin</span></code> and <code class="docutils literal notranslate"><span class="pre">2020_01_02_HighPhalloidin</span></code>).</p>
<section id="arrayed-cell-painting-experiments">
<h3>Arrayed Cell Painting experiments<a class="headerlink" href="#arrayed-cell-painting-experiments" title="Permalink to this heading">#</a></h3>
<p>Most Cell Painting experiments are arrayed, meaning that each well contains a single perturbation and therefore every cell within that well has the same perturbation.
For arrayed experiments, within each <code class="docutils literal notranslate"><span class="pre">YYYY_MM_DD_&lt;batch-name&gt;</span></code> batch subfolder there is typically an <code class="docutils literal notranslate"><span class="pre">illum</span></code> and an <code class="docutils literal notranslate"><span class="pre">images</span></code> folder.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">images</span></code> folder contains a <code class="docutils literal notranslate"><span class="pre">&lt;full-plate-name&gt;</span></code> folder for each plate imaged in that batch.
The structure beneath the <code class="docutils literal notranslate"><span class="pre">&lt;full-plate-name&gt;</span></code> folder depends on your imager, but it should contain all the images from the plate, and perhaps some other related metadata generated by the imager.
Note that the microscope used for many datasets within the Cell Painting Gallery creates an <code class="docutils literal notranslate"><span class="pre">Images</span></code> folder nested below the <code class="docutils literal notranslate"><span class="pre">&lt;full-plate-name&gt;</span></code> folder - this is considered imager-specific and should not be confused with the two higher-level <code class="docutils literal notranslate"><span class="pre">images</span></code> folder that are a part of the required folder nesting.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">illum</span></code> folder contains a <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;</span></code> folder for each plate imaged in that batch.
The <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;</span></code> can match the <code class="docutils literal notranslate"><span class="pre">&lt;full-plate-name&gt;</span></code> or it can be truncated if the <code class="docutils literal notranslate"><span class="pre">&lt;full-plate-name&gt;</span></code> is long.
Note that the relationship between <code class="docutils literal notranslate"><span class="pre">&lt;full-plate-name&gt;</span></code> and <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;</span></code> needs to be immediately obvious and the <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;</span></code> still needs to be a unique identifier (e.g. <code class="docutils literal notranslate"><span class="pre">full-plate-name</span></code> is <code class="docutils literal notranslate"><span class="pre">BR00117035__2021-05-02T16_02_51-Measurement1</span></code> and <code class="docutils literal notranslate"><span class="pre">plate-name</span></code> is <code class="docutils literal notranslate"><span class="pre">BRO00127035</span></code>).
Additionally, the <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;</span></code> used in the <code class="docutils literal notranslate"><span class="pre">images</span></code> folder must match that used in the <code class="docutils literal notranslate"><span class="pre">workspace</span></code> folder.
Within each <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;</span></code> folder there are illumination correction functions for all channels imaged in that plate, as generated by CellProfiler.
The illumination correction functions are named <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;_Illum&lt;Channel&gt;.npy</span></code>.</p>
<p>Note that the <code class="docutils literal notranslate"><span class="pre">images</span></code> folder contains the raw images as they come off of the microscope.
Though images undergo manipulation before analysis (e.g. application of illumination correction functions), intermediate, processed images are not typically saved or uploaded.
However, all of the information necessary to replicate the manipulation should be found in the <code class="docutils literal notranslate"><span class="pre">&lt;project-specific-nesting&gt;</span></code> folder (for a typical arrayed Cell Painting experiment this is just the illumination correction function).
For atypical experiments in which the images undergo more extensive manipulation and for which replicating those manipulations is challenging or prohibitive, additional folders of images may be uploaded.
Those folders will follow the format <code class="docutils literal notranslate"><span class="pre">images_&lt;manipulation-description&gt;</span></code>.
The most common example of atypical image manipulation in an arrayed experiment is when images are acquired in z-stacks and then max-projected and saved as <code class="docutils literal notranslate"><span class="pre">images_projected</span></code> before undergoing analysis.</p>
<p>An example of what this looks like in practice for a typical arrayed Cell Painting experiment is below.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cellpainting-gallery
└── cpg0016-jump
    └── source_1
    └── source_2
    └── source_3
    └── source_4
        ├── images
        │   ├── 2021_04_26_Batch1
        │   │   ├── illum
        │   │   │   ├── BR00117035
        │   │   │   │   ├── BR00117035_IllumAGP.npy
        │   │   │   │   ├── BR00117035_IllumBrightfield.npy
        │   │   │   │   ├── BR00117035_IllumBrightfield_H.npy
        │   │   │   │   ├── BR00117035_IllumBrightfield_L.npy
        │   │   │   │   ├── BR00117035_IllumDNA.npy
        │   │   │   │   ├── BR00117035_IllumER.npy
        │   │   │   │   ├── BR00117035_IllumMito.npy
        │   │   │   │   └── BR00117035_IllumRNA.npy
        │   │   │   └── BR00117036
        │   │   └── images
        │   │       ├── BR00117035__2021-05-02T16_02_51-Measurement1
        │   │       └── BR00117036__2021-05-02T18_01_40-Measurement1
        │   └── 2021_05_31_Batch2
        └── workspace
</pre></div>
</div>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">cpg0016-jump</span></code> is the project folder.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">source_4</span></code> is the anonymized nesting folder, representing Broad’s data. Note that there are multiple sources in this project, though a nesting folder is still required even if your project doesn’t have multiple sources.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">2021_04_26_Batch1</span></code> is the batch folder. Note that there are multiple batches of data acquired on different days in this project.</p></li>
<li><p>There are two plates in this example. <code class="docutils literal notranslate"><span class="pre">BR00117035__2021-05-02T16_02_51-Measurement1</span></code> is the plate name as it comes off the microscope. This naming may differ with different microscopes and different acquisition configurations.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">BR00117035</span></code> is the truncated plate name that we have given to <code class="docutils literal notranslate"><span class="pre">BR00117035__2021-05-02T16_02_51-Measurement1</span></code> that is used for naming the plate in the <code class="docutils literal notranslate"><span class="pre">illum</span></code> folder (and the <code class="docutils literal notranslate"><span class="pre">workspace</span></code> folder, discussed below).</p></li>
<li><p>In the <code class="docutils literal notranslate"><span class="pre">illum</span></code> folder, within the <code class="docutils literal notranslate"><span class="pre">BR00117035</span></code> plate folder, there are 8 separate illumination correction functions, one for each of the 8 channels imaged in that plate (e.g. <code class="docutils literal notranslate"><span class="pre">BR00117035_IllumAGP.npy</span></code> is the correction function for the AGP channel.)</p></li>
</ul>
</section>
<section id="pooled-cell-painting-experiments">
<h3>Pooled Cell Painting experiments<a class="headerlink" href="#pooled-cell-painting-experiments" title="Permalink to this heading">#</a></h3>
<p>Some Cell Painting experiments are pooled, meaning that each well contains multiple perturbations and therefore the identity of a cell’s perturbation requires additional disambiguation - typically through optically reading a barcode
For pooled experiments, within each <code class="docutils literal notranslate"><span class="pre">YYYY_MM_DD_&lt;batch-name&gt;</span></code> batch subfolder there are at least an <code class="docutils literal notranslate"><span class="pre">illum</span></code> and an <code class="docutils literal notranslate"><span class="pre">images</span></code> folder, though likely more given that the complexity of handling pooled experiments often requires generating intermediate images.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">images</span></code> folder contains the raw images as they come off of the microscope.
The <code class="docutils literal notranslate"><span class="pre">images</span></code> folder contains a <code class="docutils literal notranslate"><span class="pre">&lt;full-plate-name&gt;</span></code> folder for each plate imaged in that batch.
Within each <code class="docutils literal notranslate"><span class="pre">&lt;full-plate-name&gt;</span></code> folder are subfolders for each cycle of imaging - one or many rounds of Cell Painting and each of the rounds barcode acquisition.
Because barcoding and Cell Painting images can be acquired at different magnifications, the cycle folders start with the magnification at acquisition (e.g. <code class="docutils literal notranslate"><span class="pre">10X_</span></code> or <code class="docutils literal notranslate"><span class="pre">20X_</span></code>).
Barcoding cycle folders should end with <code class="docutils literal notranslate"><span class="pre">_SBS-&lt;n&gt;</span></code> where n is the barcoding cycle (e.g. <code class="docutils literal notranslate"><span class="pre">_SBS-1</span></code> or <code class="docutils literal notranslate"><span class="pre">_SBS-12</span></code>).
Cell Painting folders should contain <code class="docutils literal notranslate"><span class="pre">_CP_</span></code> to indicate they are Cell Painting folders and if  multiple cycles of Cell Painting are done then this should be indicated in the folder name as well (e.g. <code class="docutils literal notranslate"><span class="pre">round_1</span></code> or <code class="docutils literal notranslate"><span class="pre">round_2</span></code>).
The internal structure of the cycle round folders depends on your imager, but it should contain all the images from the plate, and perhaps some other related metadata generated by the imager.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">illum</span></code> folder contains a <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;</span></code> folder for each plate imaged in that batch.
The <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;</span></code> can match the <code class="docutils literal notranslate"><span class="pre">&lt;full-plate-name&gt;</span></code> or it can be truncated if the <code class="docutils literal notranslate"><span class="pre">&lt;full-plate-name&gt;</span></code> is long.
Note that the relationship between <code class="docutils literal notranslate"><span class="pre">&lt;full-plate-name&gt;</span></code> and <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;</span></code> needs to be immediately obvious and the <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;</span></code> still needs to be a unique identifier (e.g. <code class="docutils literal notranslate"><span class="pre">full-plate-name</span></code> is <code class="docutils literal notranslate"><span class="pre">BR00117035__2021-05-02T16_02_51-Measurement1</span></code> and <code class="docutils literal notranslate"><span class="pre">plate-name</span></code> is <code class="docutils literal notranslate"><span class="pre">BR00117035</span></code>).
Additionally, the <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;</span></code> used in the <code class="docutils literal notranslate"><span class="pre">images</span></code> folder must match that used in the <code class="docutils literal notranslate"><span class="pre">workspace</span></code> folder.
Within each <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;</span></code> folder there are illumination correction functions for all channels imaged in that plate, as generated by CellProfiler.
For Cell Painting images, the illumination correction functions are named <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;_Illum&lt;Channel&gt;.npy</span></code>.
If multiple rounds of Cell Painting occurred and there was a duplication of a painting channel between rounds, append <code class="docutils literal notranslate"><span class="pre">_Round&lt;n&gt;</span></code> to the end of the illum name (e.g. <code class="docutils literal notranslate"><span class="pre">BR00117035_IllumDNA_Round1</span></code>).
For Barcoding images, the illumination correction functions are named <code class="docutils literal notranslate"><span class="pre">&lt;plate-name&gt;_Cycle&lt;n&gt;_Illum&lt;channel&gt;.npy</span></code> where the channel is the nucleotide it corresponds to (<code class="docutils literal notranslate"><span class="pre">A</span></code>, <code class="docutils literal notranslate"><span class="pre">C</span></code>, <code class="docutils literal notranslate"><span class="pre">T</span></code>, or <code class="docutils literal notranslate"><span class="pre">G</span></code>) or the label used for alignment (e.g. <code class="docutils literal notranslate"><span class="pre">DNA</span></code>).</p>
<p>Because Pooled Cell Painting experiments may require more extensive image manipulation and replicating those manipulations is challenging or prohibitive, additional folders of images may be uploaded.
Those folders will follow the format <code class="docutils literal notranslate"><span class="pre">images_&lt;manipulation-description&gt;</span></code>.
The substructure of non-standard folders may vary greatly based on the specific workflow.
Some examples are shown below.</p>
<p>An example of what this looks like in practice for an example pooled Cell Painting experiment is below.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cellpainting-gallery
└── cpg0021-periscope
    └── broad
        ├── images
        │   ├── 20200805_A549_WG_Screen/
        │   └── 20210422_6W_CP257
        │      ├── illum
        │      │   ├── CP257A
        │      │   │   ├── CP257A_IllumAGP.npy
        │      │   │   ├── CP257A_IllumDNA.npy
        │      │   │   ├── CP257A_IllumER.npy
        │      │   │   ├── CP257A_IllumMito.npy
        │      │   │   ├── CP257A_IllumRNA.npy
        │      │   │   ├── CP257A_Cycle1_IllumA.npy
        │      │   │   ├── CP257A_Cycle1_IllumC.npy
        │      │   │   ├── CP257A_Cycle1_IllumDNA.npy
        │      │   │   ├── CP257A_Cycle1_IllumG.npy
        │      │   │   └── CP257A_Cycle1_IllumT.npy
        │      │   └── CP257B
        │      ├── images_aligned
        │      └── images_corrected_cropped
        │      │    ├── CP257A-Well1-0
        │      │    │    ├── CorrDNA
        │      │    │    │    ├──CorrDNA_Site_1.tiff
        │      │    │    │    └──CorrDNA_Site_10.tiff
        │      │    │    └── Cycle01_A
        │      │    └── CP257A-Well1-1
        │      └── images_corrected
        │      │    ├── barcoding
        │      │    │    ├── CP257A-Well1-0
        │      │    │    │    ├──Plate_CP257A_Well_1_Site_0_Cycle01_A.tiff
        │      │    │    │    └──Plate_CP257A_Well_1_Site_0_Cycle01_C.tiff
        │      │    │    └── CP257A-Well1-1
        │      │    └── painting
        │      └── images
        │          ├── CP257A
        │          │   ├── 10X_c1-SBS-1
        │          |   │    ├──Well1_Point1_0000_ChannelDAPI,Cy3,A594,Cy5,Cy7_Seq0000.nd2
        │          |   │    └──Well1_Point1_0001_ChannelDAPI,Cy3,A594,Cy5,Cy7_Seq0001.nd2
        │          │   ├── 10X_c2-SBS-2
        │          │   └── 20X_CP_CP257A
        │          └── CP257B
        └── workspace
</pre></div>
</div>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">cpg0021-periscope</span></code> is the project folder.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">broad</span></code> is source of the data.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">20210422_6W_CP257</span></code> is the batch folder. Note that there are multiple batches of data acquired on different days in this project.</p></li>
<li><p>There are two plates in this example, <code class="docutils literal notranslate"><span class="pre">CP257A</span></code> and <code class="docutils literal notranslate"><span class="pre">CP257B</span></code>.</p></li>
<li><p>In the <code class="docutils literal notranslate"><span class="pre">illum</span></code> folder, within the <code class="docutils literal notranslate"><span class="pre">CP257A</span></code> plate folder, there are many illumination correction functions, one for each of the channels imaged during Cell Painting and one for each plate, channel, and cycle of barcode acquisition.</p></li>
<li><p>The <code class="docutils literal notranslate"><span class="pre">images_corrected_cropped</span></code> folder follows one example custom structure of nesting by Plate-Well-Site, then Channel/Cycle.</p></li>
<li><p>The <code class="docutils literal notranslate"><span class="pre">images_corrected</span></code> folder follows a different example custom structure of nesting by arm, then Plate-Well-Site.</p></li>
<li><p>The <code class="docutils literal notranslate"><span class="pre">images</span></code> folder is has two example plates.
Within the <code class="docutils literal notranslate"><span class="pre">CP257A</span></code> plate folder we show folders for two cycles of barcode acquisition at 10X magnification (e.g. <code class="docutils literal notranslate"><span class="pre">10X_c1-SBS-1</span></code> and <code class="docutils literal notranslate"><span class="pre">10X_c2-SBS-2</span></code>) and one round of Cell Painting at 20X magnification (e.g. <code class="docutils literal notranslate"><span class="pre">20X_CP_CP257A</span></code>).</p></li>
</ul>
</section>
</section>
<section id="workspace-folder-structure">
<h2><code class="docutils literal notranslate"><span class="pre">workspace</span></code> folder structure<a class="headerlink" href="#workspace-folder-structure" title="Permalink to this heading">#</a></h2>
<p>Let’s look under the <code class="docutils literal notranslate"><span class="pre">workspace</span></code> folder.
Everything but images lives here.
These folders are produced when following the data processing steps in the <a class="reference external" href="https://cytomining.github.io/profiling-handbook/">Image-based Profiling Handbook</a>.
Below are the minimally required top-level folders under <code class="docutils literal notranslate"><span class="pre">workspace</span></code>.
Note that some experiments may generate additional categories of data/metadata and these should be uploaded to the <code class="docutils literal notranslate"><span class="pre">workspace</span></code> folder in their own folder/s.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cellpainting-gallery/
└── cpg0016-jump
    └── source_4
        ├── images
        └── workspace
            ├── analysis
            ├── backend
            ├── load_data_csv
            ├── metadata
            └── profiles
</pre></div>
</div>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">analysis</span></code>: contains the CSV files and optionally object outline PNGs generated by CellProfiler</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">backend</span></code>: contains the single-cell SQLite files (one per plate), the well-level aggregated profiles CSV files (also one per plate)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">load_data_csv</span></code>: contains <a class="reference external" href="https://cytomining.github.io/profiling-handbook/setup-images.html#create-loaddata-csvs">LoadData CSV files</a> used by CellProfiler to process the data</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">metadata</span></code>: contains metadata files used to annotate the profiles</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">profiles</span></code>: contains a set of well-level profiles files (one set per plate). The set comprises different stages of the CSV files produced when running the profiling recipe, as well as other output.</p></li>
</ul>
<p>Examples of additional optional folders you may upload to <code class="docutils literal notranslate"><span class="pre">workspace</span></code> include:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">assaydev</span></code>: work use to test/optimize segmentation parameters</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">embeddings</span></code>: embeddings generated from deep learning models</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">pipelines</span></code>: the CellProfiler .cppipe or .cpproj files used</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">qc</span></code>: quality control data</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">segmentation</span></code>: optimized segmentations generated independently of the <code class="docutils literal notranslate"><span class="pre">analysis</span></code> pipeline</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">software</span></code>: scripts used while handling the batch</p></li>
</ul>
<section id="analysis-folder-structure">
<h3><code class="docutils literal notranslate"><span class="pre">analysis</span></code> folder structure<a class="headerlink" href="#analysis-folder-structure" title="Permalink to this heading">#</a></h3>
<p>Within the <code class="docutils literal notranslate"><span class="pre">analysis</span></code> folder, is a folder for each batch and within each batch folder is a folder for each plate.
Within the plate folder is an additional <code class="docutils literal notranslate"><span class="pre">analysis</span></code> folder.
It is the only folder at this level; it is redundant and somewhat confusingly-named but we have kept it for legacy reasons.</p>
<p>Within the nested <code class="docutils literal notranslate"><span class="pre">analysis</span></code> folder, data is typically saved in <code class="docutils literal notranslate"><span class="pre">&lt;plate&gt;-&lt;well&gt;-&lt;site&gt;</span></code> subfolders with a .csv for each object measured (e.g. <code class="docutils literal notranslate"><span class="pre">Cells.csv</span></code>) and for experimental details (<code class="docutils literal notranslate"><span class="pre">Experiment.csv</span></code>) and whole image measurements (<code class="docutils literal notranslate"><span class="pre">Image.csv</span></code>) from that single site.
However, the grouping can vary depending on how the grouping was performed for the CellProfiler run (e.g. an experiment grouped by well instead of site would generate <code class="docutils literal notranslate"><span class="pre">&lt;plate&gt;-&lt;well&gt;</span></code> folders with the .csvs containing all of the data from the well in each .csv).</p>
<p>Often there is an additional folder such as <code class="docutils literal notranslate"><span class="pre">outlines</span></code> that contains object outlines or <code class="docutils literal notranslate"><span class="pre">masks</span></code> containing object masks.
These are the actual masks/outlines that were generated during the CellProfiler <code class="docutils literal notranslate"><span class="pre">analysis</span></code> run and used for analysis.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── analysis
    ├── 2021_04_26_Batch1
    │   ├── BR00117035
    │   │   └── analysis
    │   │       ├── BR00117035-A01-1
    │   │       │   ├── Cells.csv
    │   │       │   ├── Cytoplasm.csv
    │   │       │   ├── Experiment.csv
    │   │       │   ├── Image.csv
    │   │       │   ├── Nuclei.csv
    │   │       │   └── outlines
    │   │       │       ├── A01_s1--cell_outlines.png
    │   │       │       └── A01_s1--nuclei_outlines.png
    │   │       └── BR00117035-A01-2
    │   └── BR00117036
    └── 2021_05_31_Batch2
</pre></div>
</div>
<p>In this example batch:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">2021_04_26_Batch1</span></code> is the batch and <code class="docutils literal notranslate"><span class="pre">BR00117035</span></code> is the plate</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">BR00117035-A01-1</span></code> is a folder containing CSV files and outline files for site <code class="docutils literal notranslate"><span class="pre">1</span></code> in well <code class="docutils literal notranslate"><span class="pre">A01</span></code> in plate <code class="docutils literal notranslate"><span class="pre">BR00117035</span></code>.
Less-granular folders are acceptable as well. e.g., <code class="docutils literal notranslate"><span class="pre">BR00117035-A01</span></code> containing CSV files for the whole well and outline files for each site in the well.</p></li>
</ul>
</section>
<section id="backend-folder-structure">
<h3><code class="docutils literal notranslate"><span class="pre">backend</span></code> folder structure<a class="headerlink" href="#backend-folder-structure" title="Permalink to this heading">#</a></h3>
<p>Within the <code class="docutils literal notranslate"><span class="pre">backend</span></code> folder, is a folder for each batch and within each batch folder is a folder for each plate.
Within each plate folder is a single-cell SQLite file, comprising all measurements from all cells in the plate, and a CSV that aggregates the single-cell data into a per-well measurement.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── backend
    └── 2021_04_26_Batch1
        ├── BR00117035
        │   ├── BR00117035.csv
        │   └── BR00117035.sqlite
        └── BR00117036
</pre></div>
</div>
<p>In this example batch:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">2021_04_26_Batch1</span></code> is the batch and <code class="docutils literal notranslate"><span class="pre">BR00117035</span></code> is the plate</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">BR00117035.sqlite</span></code> is the single-cell SQLite file</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">BR00117035.csv</span></code> is the aggregated CSV file</p></li>
</ul>
</section>
<section id="load-data-csv-folder-structure">
<h3><code class="docutils literal notranslate"><span class="pre">load_data_csv</span></code> folder structure<a class="headerlink" href="#load-data-csv-folder-structure" title="Permalink to this heading">#</a></h3>
<p>Within the <code class="docutils literal notranslate"><span class="pre">load_data_csv</span></code> folder is a folder for each batch and within each batch folder is a folder for each plate.
Within the plate folder there are typically two files - a <code class="docutils literal notranslate"><span class="pre">load_data.csv</span></code> for pipelines that do not use an illumination correction function and a <code class="docutils literal notranslate"><span class="pre">load_data_with_illum.csv</span></code> for pipelines that do use an illumination correction function, however atypical workflows can have other arrangements such as a separate CSV for each pipeline in the workflow.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">load_data.csv</span></code> maps the actual file names and paths and their metadata (e.g. channel number, channel name) to the naming information passed to CellProfiler for running the images in a CellProfiler pipeline.
More information on <code class="docutils literal notranslate"><span class="pre">load_data.csv</span></code>’s and their contents is available in <a class="reference external" href="https://cellprofiler-manual.s3.amazonaws.com/CellProfiler-4.2.6/modules/fileprocessing.html#loaddata">CellProfiler documentation</a>.</p>
<p>Though CellProfiler suports multiple formats for file paths, all <code class="docutils literal notranslate"><span class="pre">load_data.csv</span></code> in the CPG use the format of column names <code class="docutils literal notranslate"><span class="pre">URL_&lt;ChannelName&gt;</span></code> where the URL starts <code class="docutils literal notranslate"><span class="pre">s3://cellpainting-gallery</span></code>. e.g.</p>
<table class="table">
<thead>
<tr class="row-odd"><th class="head"><p>URL_OrigDNA</p></th>
<th class="head"><p>URL_OrigER</p></th>
<th class="head"><p>Metadata_Plate</p></th>
<th class="head"><p>Metadata_Well</p></th>
<th class="head"><p>Metadata_Site</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>s3://cellpainting-gallery/cpg0000-jump-pilot/broad/images/2020_11_04_CPJUMP1/images/BR00116991__2020-11-05T19_51_35-Measurement1/Images/r01c01f01p01-ch5sk1fk1fl1.tiff</p></td>
<td><p>s3://cellpainting-gallery/cpg0000-jump-pilot/broad/images/2020_11_04_CPJUMP1/images/BR00116991__2020-11-05T19_51_35-Measurement1/Images/r01c01f01p01-ch4sk1fk1fl1.tiff</p></td>
<td><p>BR00116991</p></td>
<td><p>A01</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><p>s3://cellpainting-gallery/cpg0000-jump-pilot/broad/images/2020_11_04_CPJUMP1/images/BR00116991__2020-11-05T19_51_35-Measurement1/Images/r01c01f02p01-ch5sk1fk1fl1.tiff</p></td>
<td><p>s3://cellpainting-gallery/cpg0000-jump-pilot/broad/images/2020_11_04_CPJUMP1/images/BR00116991__2020-11-05T19_51_35-Measurement1/Images/r01c01f02p01-ch4sk1fk1fl1.tiff</p></td>
<td><p>BR00116991</p></td>
<td><p>A01</p></td>
<td><p>2</p></td>
</tr>
</tbody>
</table>
<p>Note that at this time, CellProfiler from source (but not built) can directly use these <code class="docutils literal notranslate"><span class="pre">load_data.csv</span></code>’s to download and process CPG images but you must have AWS credentials as it does not support unsigned requests.
<a class="reference external" href="https://github.com/DistributedScience/Distributed-CellProfiler">Distributed-CellProfiler</a> will soon support this formatting for file download but not reading directly off the bucket with S3FS.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── load_data_csv
     └── 2021_04_26_Batch1
         ├── BR00117035
         │   ├── load_data.csv
         │   └── load_data_with_illum.csv
         └── BR00117036
</pre></div>
</div>
</section>
<section id="metadata-folder-structure">
<h3><code class="docutils literal notranslate"><span class="pre">metadata</span></code> folder structure<a class="headerlink" href="#metadata-folder-structure" title="Permalink to this heading">#</a></h3>
<section id="arrayed-metadata">
<h4>arrayed metadata<a class="headerlink" href="#arrayed-metadata" title="Permalink to this heading">#</a></h4>
<p>The <code class="docutils literal notranslate"><span class="pre">metadata</span></code> folder has a slightly different structure from other <code class="docutils literal notranslate"><span class="pre">workspace</span></code> folders.
It complies with <a class="reference external" href="https://github.com/cytomining/pycytominer">pycytominer</a> metadata requirements.</p>
<p>Additional context and information may also be found in the <a class="reference external" href="https://github.com/cytomining/profiling-recipe#metadata-platemap-and-barcode_platemap-files">profiling recipe</a> and the <a class="reference external" href="https://cytomining.github.io/profiling-handbook/05-create-profiles.html#create-metadata-files">Image-based Profiling Handbook</a>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── metadata
     ├─── external_metadata
     |   └── external_metadata.tsv
     └── platemaps
         └── 2021_04_26_Batch1
             ├── barcode_platemap.csv
             └── platemap
                 └── OAA01.02.03.04.A.txt
</pre></div>
</div>
<p>All datasets have at least <code class="docutils literal notranslate"><span class="pre">barcode_platemap.csv</span></code> and <code class="docutils literal notranslate"><span class="pre">PLATEMAP.txt</span></code> files.</p>
<p>Within <code class="docutils literal notranslate"><span class="pre">barcode_platemap.csv</span></code>, there are two columns: <code class="docutils literal notranslate"><span class="pre">Assay_Plate_Barcode</span></code> and <code class="docutils literal notranslate"><span class="pre">Plate_Map_Name</span></code>.</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">Assay_Plate_Barcode</span></code> matches the plate name used for analysis.
This may be a full string match to the platenames as acquired off the imager and stored in the <code class="docutils literal notranslate"><span class="pre">images</span></code> folder (e.g.<code class="docutils literal notranslate"><span class="pre">BR00117035__2021-05-02T16_02_51-Measurement1</span></code>) or it may be a truncation of the full string as long as it is still a unique identifier (e.g. <code class="docutils literal notranslate"><span class="pre">BR00117035</span></code>).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Plate_Map_Name</span></code> is the name of a platemap in the <code class="docutils literal notranslate"><span class="pre">platemaps/BATCH/platemap</span></code> folder.
There may be one-to-one or many-to-one correspondence between <code class="docutils literal notranslate"><span class="pre">Assay_Plate_Barcode</span></code> and <code class="docutils literal notranslate"><span class="pre">Plate_Map_Name</span></code>.
Platemap naming can vary greatly from dataset to dataset depending upon the source and their data tracking/naming conventions.</p></li>
</ul>
<p>Within <code class="docutils literal notranslate"><span class="pre">PLATEMAP.txt</span></code> there at least <code class="docutils literal notranslate"><span class="pre">plate_map_name</span></code> and <code class="docutils literal notranslate"><span class="pre">well_position</span></code> columns and may be any additional number of metadata columns.</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">plate_map_name</span></code> matches the <code class="docutils literal notranslate"><span class="pre">Plate_Map_Name</span></code> in the <code class="docutils literal notranslate"><span class="pre">barcode_platemap.csv</span></code> and the <code class="docutils literal notranslate"><span class="pre">PLATEMAP</span></code> in the file name.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">well_position</span></code> matches the well names in the data output by CellProfiler and are typically based on raw image file naming as so are generally formatted like <code class="docutils literal notranslate"><span class="pre">A01</span></code> but may be upper or lowercase and may or may not have zero padding (e.g. <code class="docutils literal notranslate"><span class="pre">a1</span></code>, <code class="docutils literal notranslate"><span class="pre">a01</span></code>, <code class="docutils literal notranslate"><span class="pre">A1</span></code>, <code class="docutils literal notranslate"><span class="pre">A01</span></code>).</p></li>
</ul>
<p>Some datasets additionally have <code class="docutils literal notranslate"><span class="pre">external_metadata.tsv</span></code>.
These contain mapping between a perturbation identifier to other metadata using matching column names.</p>
<p>We do not currently enforce metadata harmonization beyond what is described here.
However, one can generally expect that metadata have been harmonized within a dataset.
We are currently exploring further metadata harmonization requirements and will update our documentation at the point of implementation.</p>
</section>
<section id="pooled-metadata">
<h4>pooled metadata<a class="headerlink" href="#pooled-metadata" title="Permalink to this heading">#</a></h4>
<p>For pooled experiments, the primary source of disambiguation of cellular perturbations is through barcode assignment to individual cells and not through a platemap, so the folder structure is different than for arrayed <code class="docutils literal notranslate"><span class="pre">metadata</span></code>.
The <code class="docutils literal notranslate"><span class="pre">Barcodes.csv</span></code> used for assignment is required and is assumed to be the same for each plate within a batch.
Other sources of metadata may be included, particularly if there are additional per-well or per-plate differences in metadata.
Additionally, we suggest the inclusion of the metadata.json dictionary used for image processing with the <a class="reference external" href="https://github.com/broadinstitute/pooled-cell-painting-image-processing">pooled cell painting image processing repository</a>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── metadata
   └── 2021_04_26_Batch1
        ├── Barcodes.csv
        └── metadata.json
</pre></div>
</div>
</section>
</section>
<section id="profiles-folder-structure">
<h3><code class="docutils literal notranslate"><span class="pre">profiles</span></code> folder structure<a class="headerlink" href="#profiles-folder-structure" title="Permalink to this heading">#</a></h3>
<p>Within the <code class="docutils literal notranslate"><span class="pre">profiles</span></code> folder is a folder for each batch and within each batch folder is a folder for each plate.
Within each plate folder are many files produced by the <a class="reference external" href="https://github.com/cytomining/profiling-recipe">profiling-recipe</a> that describe single-cell morphological profiles.
For a full description of the files, see <a class="reference external" href="https://github.com/cytomining/profiling-recipe#files-generated">profiling-recipe files generated</a>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── profiles
    └── 2021_04_26_Batch1
        ├── BR00117035
        │   ├── BR00117035.csv.gz
        │   ├── BR00117035_augmented.csv.gz
        │   ├── BR00117035_normalized.csv.gz
        │   ├── BR00117035_normalized_feature_select_negcon_plate.csv.gz
        │   ├── BR00117035_normalized_feature_select_plate.csv.gz
        │   └── BR00117035_normalized_negcon.csv.gz
        └── BR00117036
</pre></div>
</div>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">2021_04_26_Batch1</span></code> is the batch and <code class="docutils literal notranslate"><span class="pre">BR00117035</span></code> is the plate</p></li>
<li><p>The .csv files undergo gzip compression to be .csv.gz files</p></li>
</ul>
</section>
<section id="quality-control-folder-structure">
<h3><code class="docutils literal notranslate"><span class="pre">quality_control</span></code> folder structure<a class="headerlink" href="#quality-control-folder-structure" title="Permalink to this heading">#</a></h3>
<p>The <code class="docutils literal notranslate"><span class="pre">quality_control</span></code> folder has the slightly different structure.
The files are all produced by the <a class="reference external" href="https://github.com/jump-cellpainting/profiling-recipe#files-generated">profiling-recipe</a>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── quality_control
    └── heatmap
        └── 2021_04_26_Batch1
            ├── BR00117035
            │   ├── BR00117035_cell_count.png
            │   ├── BR00117035_correlation.png
            │   ├── BR00117035_position_effect.png
            │   └── and possibly others
            └── BR00117036
</pre></div>
</div>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">2021_04_26_Batch1</span></code> is the batch and <code class="docutils literal notranslate"><span class="pre">BR00117035</span></code> is the plate</p></li>
</ul>
</section>
<section id="segmentation-folder-structure">
<h3><code class="docutils literal notranslate"><span class="pre">segmentation</span></code> folder structure<a class="headerlink" href="#segmentation-folder-structure" title="Permalink to this heading">#</a></h3>
<p>Files within <code class="docutils literal notranslate"><span class="pre">segmentation</span></code> are object segmentations (masks and/or outlines) generated outside of the final (usually CellProfiler) <code class="docutils literal notranslate"><span class="pre">analysis</span></code> pipeline and therefore likely differ from the segmentations used to make the dataset’s profiles.
They are considered “state-of-the-field” or better in quality.
The <code class="docutils literal notranslate"><span class="pre">model</span></code> and/or <code class="docutils literal notranslate"><span class="pre">training</span></code> data are optionally included.
At minimum, a <code class="docutils literal notranslate"><span class="pre">README.md</span></code> describing the model and where to access it is included in the <code class="docutils literal notranslate"><span class="pre">model</span></code> folder.</p>
<p>Within <code class="docutils literal notranslate"><span class="pre">objects</span></code> are the segmented objects, nested into Batch and Plate folders</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── segmentation
    └── &lt;software&gt;_&lt;hash&gt; (e.g. cellpose_c632b05b6930)
        ├── model
        │   └──README.md
        ├── training
        └── objects
            ├── 2021_04_26_Batch1
            │    ├── BR00117035
            │    └── BR00117036
            └── 2021_05_31_Batch2
</pre></div>
</div>
<p>Within Batch and Plate folders, substructure can vary.
The structure used for <code class="docutils literal notranslate"><span class="pre">cpg0016-jump</span></code> is as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── BR00117036
    ├── BR00117036.zarr
    │    └──&lt;source&gt;__&lt;batch&gt;__&lt;plate&gt;__&lt;well&gt;__&lt;site&gt;
    │       ├──label_image/
    │       ├──single_cell_data/
    │       ├──single_cell_index/
    │       └──.zgroup
    └── channel_mapping.json
</pre></div>
</div>
</section>
</section>
<section id="workspace-dl-folder-structure">
<h2><code class="docutils literal notranslate"><span class="pre">workspace_dl</span></code> folder structure<a class="headerlink" href="#workspace-dl-folder-structure" title="Permalink to this heading">#</a></h2>
<p><strong>NOTE: This section is work in progress.
More documentation will be added.
The structure may change.</strong></p>
<p>Within the <code class="docutils literal notranslate"><span class="pre">workspace_dl</span></code> folder are several subfolders for different classes of data.</p>
<p>Within the subfolders are folders for the
In this example we have used <code class="docutils literal notranslate"><span class="pre">efficientnet_v2_imagenet1k_s_feature_vector_2_ec756ff</span></code> where <code class="docutils literal notranslate"><span class="pre">efficientnet</span></code> is the name of the network, <code class="docutils literal notranslate"><span class="pre">imagenet1k</span></code> is the dataset that was used for training, and <code class="docutils literal notranslate"><span class="pre">ec756ff</span></code> is a hash for the model.
Note that it is possible to use other identifiers for the model such as a Zenodo DOI.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cellpainting-gallery/
└── cpg0016-jump
    └── source_4
        └── workspace_dl
            ├── collated
            ├── consensus
            ├── embeddings
            └── profiles
</pre></div>
</div>
<section id="collated-folder-structure">
<h3><code class="docutils literal notranslate"><span class="pre">collated</span></code> folder structure<a class="headerlink" href="#collated-folder-structure" title="Permalink to this heading">#</a></h3>
<p>The <code class="docutils literal notranslate"><span class="pre">collated</span></code> contains .csv or .parquet files with well-level profiles for all plates in a folder for each network/model.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── collated
        └── efficientnet_v2_imagenet1k_s_feature_vector_2_ec756ff
            └── collated.parquet
</pre></div>
</div>
</section>
<section id="consensus-folder-structure">
<h3><code class="docutils literal notranslate"><span class="pre">consensus</span></code> folder structure<a class="headerlink" href="#consensus-folder-structure" title="Permalink to this heading">#</a></h3>
<p>The <code class="docutils literal notranslate"><span class="pre">consensus</span></code> folder contains .csv or .parquet files with treatment-level profiles for all plates in a folder for each network/model.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── consensus
        └── efficientnet_v2_imagenet1k_s_feature_vector_2_ec756ff
            └── consensus.parquet
</pre></div>
</div>
</section>
<section id="embeddings-folder-structure">
<h3><code class="docutils literal notranslate"><span class="pre">embeddings</span></code> folder structure<a class="headerlink" href="#embeddings-folder-structure" title="Permalink to this heading">#</a></h3>
<p>The <code class="docutils literal notranslate"><span class="pre">embeddings</span></code> folder contains a subfolder for each network/model, with subfolders for each batch. Within each batch folder is a subfolder for each plate. Within each plate subfolder is a subfolder for well-site. In the well-site subfolder is a .npz or .parquet file with single-cell features extracted from the single image.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── embeddings
        └── efficientnet_v2_imagenet1k_s_feature_vector_2_ec756ff
            ├── 2021_04_26_Batch1
            │   ├── BR00117035
            │   │       ├── A01-1
            │   │       │   └── embedding.parquet
            │   │       └── A01-2
            │   └── BR00117036
            └── 2021_05_31_Batch2
</pre></div>
</div>
<p>In this example batch:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">2021_04_26_Batch1</span></code> is the batch and <code class="docutils literal notranslate"><span class="pre">BR00117035</span></code> is the plate</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">efficientnet_v2_imagenet1k_s_feature_vector_2_ec756ff</span></code> is an identifier for the deep learning network, suffixed with some hash for the model</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">A01-1</span></code> is a folder containing the embedding file for site <code class="docutils literal notranslate"><span class="pre">1</span></code> in well <code class="docutils literal notranslate"><span class="pre">A01</span></code> in plate <code class="docutils literal notranslate"><span class="pre">BR00117035</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">embedding.parquet</span></code> is the single-cell Parquet file containing the embeddings</p></li>
</ul>
<p>The folder structure is a little different for DeepProfiler-generated output in that the well-site subfolder is replaced by a well subfolders with subfolders per site.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── embeddings
        └── efficientnet_v2_imagenet1k_s_feature_vector_2_ec756ff
            ├── 2021_04_26_Batch1
            │   ├── BR00117035
            │   │       ├── A01
            │   │       │   └── 1
            │   │       │   │   └── embedding.npz
            │   │       │   └── 2
            │   └── BR00117036
            └── 2021_05_31_Batch2
</pre></div>
</div>
</section>
<section id="id1">
<h3><code class="docutils literal notranslate"><span class="pre">profiles</span></code> folder structure<a class="headerlink" href="#id1" title="Permalink to this heading">#</a></h3>
<p>Within the <code class="docutils literal notranslate"><span class="pre">profiles</span></code> folder is a folder for the deep learning network with its hash.
Within the network folder is a folder for each batch and within each batch folder is a file for each plate.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>└── profiles
        └── efficientnet_v2_imagenet1k_s_feature_vector_2_ec756ff
            ├── 2021_04_26_Batch1
            │   ├── BR00117035
            │   │       └── BR00117035.parquet
            │   └── BR00117036
            └── 2021_05_31_Batch2
</pre></div>
</div>
</section>
</section>
<section id="complete-folder-structure">
<h2>Complete folder structure<a class="headerlink" href="#complete-folder-structure" title="Permalink to this heading">#</a></h2>
<p>Here’s the complete folder structure for a sample project with CellProfiler-based features.</p>
<details>
 <summary>Click here</summary>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span> └── cellpainting-gallery
    └── cpg0016-jump
        └── source_4
            ├── images
            │   ├── 2021_04_26_Batch1
            │   │   ├── illum
            │   │   │   ├── BR00117035
            │   │   │   │   ├── BR00117035_IllumAGP.npy
            │   │   │   │   ├── BR00117035_IllumBrightfield.npy
            │   │   │   │   ├── BR00117035_IllumBrightfield_H.npy
            │   │   │   │   ├── BR00117035_IllumBrightfield_L.npy
            │   │   │   │   ├── BR00117035_IllumDNA.npy
            │   │   │   │   ├── BR00117035_IllumER.npy
            │   │   │   │   ├── BR00117035_IllumMito.npy
            │   │   │   │   └── BR00117035_IllumRNA.npy
            │   │   │   └── BR00117036
            │   │   └── images
            │   │       ├── BR00117035__2021-05-02T16_02_51-Measurement1
            │   │       └── BR00117036__2021-05-02T18_01_40-Measurement1
            │   └── 2021_05_31_Batch2
            └── workspace
                ├── analysis
                │   ├── 2021_04_26_Batch1
                │   │   ├── BR00117035
                │   │   │   └── analysis
                │   │   │       ├── BR00117035-A01-1
                │   │   │       │   ├── Cells.csv
                │   │   │       │   ├── Cytoplasm.csv
                │   │   │       │   ├── Image.csv
                │   │   │       │   ├── Nuclei.csv
                │   │   │       │   └── outlines
                │   │   │       │       ├── A01_s1--cell_outlines.png
                │   │   │       │       └── A01_s1--nuclei_outlines.png
                │   │   │       └── BR00117035-A01-2
                │   │   └── BR00117036
                │   └── 2021_05_31_Batch2
                ├── backend
                │   └── 2021_04_26_Batch1
                │       ├── BR00117035
                │       │   ├── BR00117035.csv
                │       │   └── BR00117035.sqlite
                │       └── BR00117036
                ├── load_data_csv
                │   └── 2021_04_26_Batch1
                │       ├── BR00117035
                │       │   ├── load_data.csv.gz
                │       │   └── load_data_with_illum.csv.gz
                │       └── BR00117036
                ├── metadata
                │   ├─── external_metadata
                |   |   └── external_metadata.tsv
                │   └── platemaps
                |       └── 2021_04_26_Batch1
                |           ├── platemap
                |           │   └── OAA01.02.03.04.A.txt
                |           └── barcode_platemap.csv
                ├── quality_control
                │   └── heatmap
                │       └── 2021_04_26_Batch1
                │           ├── BR00117035
                │           │   ├── BR00117035_cell_count.png
                │           │   ├── BR00117035_correlation.png
                │           │   ├── BR00117035_position_effect.png
                │           │   └── and possibly others
                │           └── BR00117036
                └── profiles
                    └── 2021_04_26_Batch1
                        ├── BR00117035
                        │   ├── BR00116991_augmented.csv.gz
                        │   ├── BR00116991_normalized.csv.gz
                        │   ├── BR00116991_normalized_feature_select_negcon_plate.csv.gz
                        │   ├── BR00116991_normalized_feature_select_plate.csv.gz
                        │   ├── BR00116991_normalized_negcon.csv.gz
                        │   ├── BR00117035.csv.gz
                        │   └── and others https://github.com/cytomining/profiling-recipe#files-generated
                        └── BR00117036
</pre></div>
</div>
 </details>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="overview.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Cell Painting Gallery Overview</p>
      </div>
    </a>
    <a class="right-next"
       href="feature_sets.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Feature Sets</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#images-folder-structure"><code class="docutils literal notranslate"><span class="pre">images</span></code> folder structure</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#arrayed-cell-painting-experiments">Arrayed Cell Painting experiments</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#pooled-cell-painting-experiments">Pooled Cell Painting experiments</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#workspace-folder-structure"><code class="docutils literal notranslate"><span class="pre">workspace</span></code> folder structure</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#analysis-folder-structure"><code class="docutils literal notranslate"><span class="pre">analysis</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#backend-folder-structure"><code class="docutils literal notranslate"><span class="pre">backend</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-data-csv-folder-structure"><code class="docutils literal notranslate"><span class="pre">load_data_csv</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#metadata-folder-structure"><code class="docutils literal notranslate"><span class="pre">metadata</span></code> folder structure</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#arrayed-metadata">arrayed metadata</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#pooled-metadata">pooled metadata</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#profiles-folder-structure"><code class="docutils literal notranslate"><span class="pre">profiles</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#quality-control-folder-structure"><code class="docutils literal notranslate"><span class="pre">quality_control</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#segmentation-folder-structure"><code class="docutils literal notranslate"><span class="pre">segmentation</span></code> folder structure</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#workspace-dl-folder-structure"><code class="docutils literal notranslate"><span class="pre">workspace_dl</span></code> folder structure</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#collated-folder-structure"><code class="docutils literal notranslate"><span class="pre">collated</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#consensus-folder-structure"><code class="docutils literal notranslate"><span class="pre">consensus</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#embeddings-folder-structure"><code class="docutils literal notranslate"><span class="pre">embeddings</span></code> folder structure</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1"><code class="docutils literal notranslate"><span class="pre">profiles</span></code> folder structure</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#complete-folder-structure">Complete folder structure</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Broad Institute
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=5b4479735964841361fd"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>